weasyl/searchtag.py

```diff
@@ -118,6 +118,49 @@ def parse_tags(text):
     return tags
 
 
+def parse_restricted_tags(text):
+    """
+    A custom implementation of ``parse_tags()`` for the searchtag restriction list.
+    Enforces the desired characteristics of STBL tags, and allows an asterisk
+    character, whereas ``parse_tags()`` would strip asterisks.
+
+    Parameters:
+        text: The string to parse for tags
+
+    Returns:
+        tags: A set() with valid tags.
+    """
+    tags = set()
+
+    for i in _TAG_DELIMITER.split(text):
+        target = "".join([c for c in i if ord(c) < 128])
+        target = "".join(i for i in target if i.isalnum() or i in "_*")
+        target = target.strip("_")
+        target = "_".join(i for i in target.split("_") if i)
```

Comment:
I just ran this through a test in the Python shell, and yes, that's what it appears to do.
```
kyra@kitty:~$ python
Python 2.7.9 (default, Jun 29 2016, 13:08:31)
[GCC 4.9.2] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> a = "qq___q"
>>> q = "_".join(i for i in a.split("_") if i)
>>> q
'qq_q'
```
I just did a straight copy of the ``searchtag.parse_tags()`` logic which pulls from ``define.get_search_tag()``, which itself executes the same bit of code at L524 (as of rev. f645965).